pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdProduce.java

```diff
@@ -107,6 +107,9 @@
                description = "Rate (in msg/sec) at which to produce," +
                        " value 0 means to produce messages as fast as possible.")
     private double publishRate = 0;
+
+    @Parameter(names = { "-db", "--disable-batching" }, description = "Disable batch sending of messages")
```

Comment:
It better to named `--batch-enabled`? and the default is true.

Reply:
Normally, boolean type options do not require a value, so the default value should be false.
```sh
# By default batching is enabled
$ ./bin/pulsar-client produce -m hello persistent://public/default/t1

# Batching is enabled in this case as well
$ ./bin/pulsar-client produce -m hello --batch-enabled persistent://public/default/t1
```

We can set the default value to true by specifying `arity = 1`, but it may be a bit verbose. 
https://jcommander.org/#_boolean
```sh
# Batching is enabled
$ ./bin/pulsar-client produce -m hello --batch-enabled true persistent://public/default/t1

# Batching is disabled
$ ./bin/pulsar-client produce -m hello --batch-enabled false persistent://public/default/t1
```

Reply:
Ok, that makes sense.

How about keeping the same configurations with the perf producer?

```
        @Parameter(names = { "-b",
                "--batch-time-window" }, description = "Batch messages in 'x' ms window (Default: 1ms)")
        public double batchTimeMillis = 1.0;

        @Parameter(names = {
            "-bm", "--batch-max-messages"
        }, description = "Maximum number of messages per batch")
        public int batchMaxMessages = DEFAULT_BATCHING_MAX_MESSAGES;

        @Parameter(names = {
            "-bb", "--batch-max-bytes"
        }, description = "Maximum number of bytes per batch")
        public int batchMaxBytes = 4 * 1024 * 1024
```

```
if (arguments.batchTimeMillis <= 0.0 && arguments.batchMaxMessages <= 0) {
                producerBuilder.enableBatching(false);
            }
```

Reply:
In the `pulsar-client` command, one batch message always contains only one message, even if batching is enabled. This is because we are using the synchronous method `send()` instead of the asynchronous method `sendAsync()`.

https://github.com/apache/pulsar/blob/3c770a18a807498634124161f95bc4f0888d5315/pulsar-client-tools/src/main/java/org/apache/pulsar/client/cli/CmdProduce.java#L297

So, unlike the perf producer, I don't think such fine-grained settings are useful.

Reply:
There is no particular benefit to enabling batching for the `pulsar-client` command, so we have the option of always disabling it.

Reply:
Ok, that make sense.